src/object/obj_tx.c

```diff
@@ -1305,6 +1301,12 @@ dc_tx_classify_common(struct dc_tx *tx, struct daos_cpd_sub_req *dcsr,
 		if (rc != 0)
 			goto out;
 
+		if (reasb_req != NULL && reasb_req->tgt_bitmap != NIL_BITMAP &&
+		    isclr(reasb_req->tgt_bitmap, shard->do_shard % daos_oclass_grp_size(oca))) {
```

Comment:
could you please remind me why need to change it here?
original code of "isclr(reasb_req->tgt_bitmap, idx)" does not work? as the "idx" is physical index seems original code should work?

And not a problem of this patch, but above L1270 "start_tgt = obj_ec_shard_idx(obj, dcsr->dcsr_dkey_hash, obj_get_grp_size(obj) - 1);" seems incorrect, should it be "start_tgt = obj_ec_shard_idx(obj, dcsr->dcsr_dkey_hash, obj_ec_tgt_nr(oca) - 1);" instead?

Reply:
>And not a problem of this patch, but above L1270 "start_tgt = obj_ec_shard_idx(obj, dcsr->dcsr_dkey_hash, obj_get_grp_size(obj) - 1);" seems incorrect, should it be "start_tgt = obj_ec_shard_idx(obj, dcsr->dcsr_dkey_hash, obj_ec_tgt_nr(oca) - 1);" instead?

Hmm, indeed, I will check this in a different patch.